oldsiva: add support for old siva+rooted-repository format

mcarmonaa opened the pull request:
Closes #41

jfontan commented on oldsiva/helper_test.go:
```diff
@@ -0,0 +1,126 @@
+
+*/
+
+const testDir = "_oldsiva_testdata"
```
Why `_oldsiva_testdata` instead of `_testdata`? Besides that LGTM! :tada:

mcarmonaa commented:
@jfontan I've renamed `_oldsiva_testdata` to `_testdata`. Also I changed the library constructor to dereference the `*LibraryOptions` as we have changed it in the other library implementations in #59
